src/default_thread_pool.h

```diff
@@ -0,0 +1,47 @@
+
+// Copyright 2024-present the vsag project
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <functional>
+#include <future>
+
+#include "ThreadPool.h"
+#include "vsag/options.h"
+#include "vsag/thread_pool.h"
+
+namespace vsag {
+
+class DefaultThreadPool : public ThreadPool {
+public:
+    explicit DefaultThreadPool(std::size_t threads);
+
+    std::future<void>
+    enqueue(std::function<void(void)> task) override;
+
+    void
+    wait_until_empty() override;
+
+    void
+    set_queue_size_limit(std::size_t limit) override;
+
+    void
+    set_pool_size(std::size_t limit) override;
```

Comment:
public method's name should start with a capital letter

Reply:
done